platform/ui/src/components/SidePanel/SidePanel.tsx

```diff
@@ -52,95 +51,119 @@ const classesMap = {
+const getNumGridColumns = (numTabs: number) => {
+  if (numTabs === 1) {
+    return 1;
+  }
 
-  const openStatus = panelOpen ? 'open' : 'closed';
-  const style = Object.assign({}, styleMap[openStatus][side], baseStyle);
+  // Start by calculating the number of tabs assuming each tab was accompanied by a spacer.
+  const tabWidth = getTabWidth(numTabs);
+  const gridWidth = getGridWidth(numTabs);
+  const numTabsWithOneSpacerEach = Math.floor(gridWidth / (tabWidth + tabSpacerWidth));
 
-  const ActiveComponent = tabs[activeTabIndex].content;
+  // But there is always one less spacer than tabs, so now check if an extra tab with one less spacer fits.
+  if (
+    (numTabsWithOneSpacerEach + 1) * tabWidth + numTabsWithOneSpacerEach * tabSpacerWidth <=
+    gridWidth
+  ) {
+    return numTabsWithOneSpacerEach + 1;
+  }
 
-  useEffect(() => {
-    if (panelOpen && swiper) {
-      swiper.slideTo(activeTabIndex, 500);
-    }
-  }, [panelOpen, swiper]);
+  return numTabsWithOneSpacerEach;
+};
 
-  useEffect(() => {
-    if (swiper) {
-      swiper.params.navigation.prevEl = prevRef.current;
-      swiper.params.navigation.nextEl = nextRef.current;
-      swiper.navigation.init();
-      swiper.navigation.update();
-    }
-  }, [swiper]);
+const getGridStyle = (side: string, numTabs: number = 0): CSSProperties => {
+  const gridWidth = getGridWidth(numTabs);
+  const relativePosition = Math.max(0, Math.floor(expandedWidth - gridWidth) / 2 - closeIconWidth);
+  return {
+    position: 'relative',
+    ...(side === 'left' ? { right: `${relativePosition}px` } : { left: `${relativePosition}px` }),
+    width: `${gridWidth}px`,
+  };
+};
+
+const getTabClassNames = (
+  numColumns: number,
+  numTabs: number,
+  tabIndex: number,
+  isActiveTab: boolean
+) =>
+  classnames('h-[28px] mb-[2px] cursor-pointer text-white bg-black', {
+    'hover:text-primary-active': !isActiveTab,
+    'rounded-l': tabIndex % numColumns === 0,
+    'rounded-r': (tabIndex + 1) % numColumns === 0 || tabIndex === numTabs - 1,
+  });
+
+const getTabStyle = (numTabs: number) => {
+  return {
+    width: `${getTabWidth(numTabs)}px`,
+  };
+};
+
+const getTabIconClassNames = (numTabs: number, isActiveTab: boolean) => {
+  return classnames('h-full w-full flex items-center justify-center', {
+    'bg-customblue-40': isActiveTab,
+    rounded: isActiveTab,
+  });
+};
+
+const SidePanel = ({ side, className, activeTabIndex: activeTabIndexProp, tabs, onOpened }) => {
+  const { t } = useTranslation('SidePanel');
+
+  const [panelOpen, setPanelOpen] = useState();
+  const [activeTabIndex, setActiveTabIndex] = useState(0);
+
+  const openStatus = panelOpen ? 'open' : 'closed';
+  const style = Object.assign({}, styleMap[openStatus][side], baseStyle);
+
+  const ActiveComponent = tabs[activeTabIndex]?.content;
 
   const updatePanelOpen = useCallback((panelOpen: boolean) => {
     setPanelOpen(panelOpen);
     if (panelOpen) {
-      setHasBeenOpened(true);
+      onOpened?.();
```

Comment:
I'm not sure if `onOpened` or `onOpen`, I feel like the second one match the rest of codebase better